charts/featureform/templates/db/upgrade.yaml

```diff
@@ -0,0 +1,35 @@
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+#  Copyright 2024 FeatureForm Inc.
+#
+
+{{ if eq .Values.stateProvider "psql" }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: featureform-db-migration-up
+  labels:
+    chart: featureform
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: db-migration-up
+          image: "{{ .Values.repository | default .Values.repository }}/{{ .Values.psql.migration.image.name }}:{{ .Values.versionOverride | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.pullPolicy }}
```

Comment:
probably need to include [ttlSecondsAfterFinished](https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/) so the job doesn't stop the next helm upgrade from completing, unless goose auto deletes it?

Reply:
The `"helm.sh/hook-delete-policy": hook-succeeded` annotation does, but that raises a question whether we want to also delete it if it failed? I'd think no so someone can go and debug it, but interested in hearing feedback.

Reply:
it's a good question, because we'd at least like to see the logs in case it does blow up. can it be conditional? if the job succeeds, then delete, otherwise keep the job alive so we can see the logs?

Reply:
I think always delete it, but it should fail the deployment or something right

Reply:
So to be clear, it currently is conditional (so it's deleted if it's successful bc the annotation mentioned above is applied).

It should also fail the deployment, but we'd have no logs as to why the job failed, which is a situation we'd want to avoid. I think that the current approach of deleting only successful job executions should be good, unless there's major pushback.